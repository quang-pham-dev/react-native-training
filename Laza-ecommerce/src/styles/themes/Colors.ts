const Colors = {
  // colors
  primaryColor: '#3D93F8',
  secondaryColor: '#FFC107',

  white: '#fff',
  black: '#000',
  transparent: 'transparent',
  // text√é=
  textBlack: '#1D1E20',
  textGray: '#8F959E',
  textRed: '#EA4335',
  textWhite: '#FEFEFE',
  // border
  borderGray: '#E7E8EA',
  //
  facebook: '#4267B2',
  twitter: '#1DA1F2',
  google: '#EA4335',
  scaleGray: '#FCFCFC',
  lightBlue: '#097ade',
  lightGray: '#efefef',
  tintColor: '#e91e63',
  shadowGray: 'rgba(0, 0, 0, 0.15)',
  // backgroundColor
  primaryBackground: '#8F959E',
  secondaryBackground: '#fff',
  Background: '#F5F6FA',
};

export default Colors;
