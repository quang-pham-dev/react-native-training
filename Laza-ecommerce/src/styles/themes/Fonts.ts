const size = {
  xxs: 12,
  custom: 13,
  xs: 14,
  base: 16,
  standard: 15,
  sm: 17,
  default: 18,
  lg: 28,
  xl: 31,
  xlg: 38,
};

const lineHeight = {
  xxs: 12,
  xs: 14,
  base: 16,
  sm: 18,
  lg: 22,
};

const fontFamily = {
  Inter_100Thin: 'Inter_100Thin',
  Inter_200ExtraLight: 'Inter_200ExtraLight',
  Inter_300Light: 'Inter_300Light',
  Inter_400Regular: 'Inter_400Regular',
  Inter_500Medium: 'Inter_500Medium',
  Inter_600SemiBold: 'Inter_600SemiBold',
  Inter_700Bold: 'Inter_700Bold',
  Inter_800ExtraBold: 'Inter_800ExtraBold',
  Inter_900Black: 'Inter_900Black',
};

export default {
  size,
  lineHeight,
  fontFamily,
};
