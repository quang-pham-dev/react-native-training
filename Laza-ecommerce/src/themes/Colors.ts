const Colors = {
  // colors
  primaryColor: '#3D93F8',
  secondaryColor: '#FFC107',

  white: '#fff',
  black: '#000',
  transparent: 'transparent',

  // text colors
  textBlack: '#1D1E20',
  textGray: '#8F959E',
  textRed: '#EA4335',
  textWhite: '#FEFEFE',
  textLogout: '#FF5757',

  // border colors
  borderGray: '#E7E8EA',

  // background colors
  facebook: '#4267B2',
  twitter: '#1DA1F2',
  google: '#EA4335',
  scaleGray: '#FCFCFC',
  lightGreen: '#34C759',
  lightBlue: '#097ade',
  lightGray: '#F2F2F2',
  tintColor: '#e91e63',
  shadowGray: 'rgba(0, 0, 0, 0.15)',
  primaryBackground: '#fff',
  secondaryBackground: '#F5F6FA'
};

export default Colors;
