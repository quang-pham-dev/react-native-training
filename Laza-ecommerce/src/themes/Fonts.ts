const size = {
  min: 11, // 5% text of design
  small: 13, //  10% text of design
  normal: 15, // 10% text of design
  default: 17, // 70% text design has size 17
  medium: 18,
  large: 20,
  h6: 24,
  h5: 28, // 5% text of design
  h4: 32,
  h3: 36,
  h2: 40,
  h1: 44
};

const lineHeight = {
  xxs: 12,
  xs: 14,
  base: 16,
  sm: 18,
  lg: 22,
  xl: 26,
  xxl: 30
};

const fontFamily = {
  Inter_100Thin: 'Inter_100Thin',
  Inter_200ExtraLight: 'Inter_200ExtraLight',
  Inter_300Light: 'Inter_300Light',
  Inter_400Regular: 'Inter_400Regular',
  Inter_500Medium: 'Inter_500Medium',
  Inter_600SemiBold: 'Inter_600SemiBold'
};

export default {
  size,
  lineHeight,
  fontFamily
};
