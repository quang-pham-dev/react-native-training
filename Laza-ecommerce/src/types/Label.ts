import { StyleProp, TextStyle, ViewStyle } from 'react-native';

export type labelTextProps = {
  fontSize?: number;
  labelName?: string;
  fontFamily?: string;
  color?: string;
  lineHeight: number;
  labelStyles?: StyleProp<TextStyle>;
  labelWrapperStyles?: StyleProp<ViewStyle>;
  labelType?: string;
  hasBorder?: boolean;
};
