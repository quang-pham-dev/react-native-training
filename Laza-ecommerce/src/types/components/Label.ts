import { StyleProp, TextStyle, ViewStyle } from 'react-native';

export interface ILabelTextProps {
  fontSize?: number;
  labelName?: string;
  fontFamily?: string;
  color?: string;
  lineHeight?: number;
  labelStyles?: StyleProp<TextStyle>;
  labelWrapperStyles?: StyleProp<ViewStyle>;
  labelType?: string;
  hasBorder?: boolean;
}
