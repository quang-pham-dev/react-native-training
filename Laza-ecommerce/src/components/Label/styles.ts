import { StyleSheet } from 'react-native';
// Themes
import { Colors, Fonts, Metrics } from 'styles/themes';

const styles = StyleSheet.create({
  labelText: {
    fontSize: Fonts.size.sm,
    color: Colors.textBlack,
  },

  labelContainerBorder: {
    borderWidth: 1,
    borderColor: Colors.textGray,
    backgroundColor: Colors.white,
  },

  labelContainer: {
    borderRadius: Metrics.borderRadius.roundedBase,
    paddingVertical: Metrics.padding.sm,
    paddingHorizontal: Metrics.padding.base,
    backgroundColor: Colors.primaryBackground,
  },
});

export default styles;
