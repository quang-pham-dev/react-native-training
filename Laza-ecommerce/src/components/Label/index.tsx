import React, { memo } from 'react';
import { Text, View } from 'react-native';
import isEqual from 'react-fast-compare';

// Types
import { ILabelTextProps } from 'types/components/Label';

// Styles
import styles from './styles';

const Label = ({
  labelName,
  fontSize,
  fontFamily,
  color,
  labelWrapperStyles,
  lineHeight,
}: ILabelTextProps) => {
  const labelStyles = {
    fontSize: fontSize,
    fontFamily: fontFamily,
    color,
    ...(lineHeight && { lineHeight }),
  };
  return (
    <View style={labelWrapperStyles}>
      <Text style={[styles.labelText, labelStyles]}>{labelName}</Text>
    </View>
  );
};

export default memo(Label, isEqual);
