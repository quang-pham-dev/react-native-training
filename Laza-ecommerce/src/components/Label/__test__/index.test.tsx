import React from 'react';

// LIBS
import renderer from 'react-test-renderer';

// Components
import Label from 'components/Label';

// Themes
import Colors from 'themes/Colors';
import Fonts from 'themes/Fonts';

describe('Label Component', () => {
  const props = {
    labelName: 'Label Name',
    fontSize: 17,
    fontFamily: Fonts.fontFamily.Inter_400Regular,
    color: Colors.textBlack,
    lineHeight: 15,
  };

  test('should render correctly', () => {
    const tree = renderer.create(<Label {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
