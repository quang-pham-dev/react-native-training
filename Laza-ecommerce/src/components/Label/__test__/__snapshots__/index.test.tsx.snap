// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component should render correctly 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#1D1E20",
          "fontSize": 17,
        },
        Object {
          "color": "#1D1E20",
          "fontFamily": "Inter_400Regular",
          "fontSize": 17,
          "lineHeight": 15,
        },
      ]
    }
  >
    Label Name
  </Text>
</View>
`;
